os/mm/mm_heap/mm_mallinfo.c

```diff
@@ -99,6 +213,10 @@ int mm_mallinfo(FAR struct mm_heap_s *heap, FAR struct mallinfo *info)
 #define region 0
 #endif
 
+	if (mm_check_heap_corruption(heap) != OK) {
+		DEBUGASSERT(0);
```

Comment:
DEBUGASSERT is meaningful when CONFIG_DEBUG is enabled. When it is disabled, DEBUGASSERT is the same as nothing. I think this case means heap is already corrupted. So ASSERT is right. not debugassert.

Reply:
OK. Done. 
I was under the assumption that we can let the system run even if heap is corrupted, until it causes some assert by accessing the corrupted node. But, I agree that it will be difficult to debug in such case. So, its good to change it to ASSERT.